sdv/metadata/table.py

```diff
@@ -398,15 +398,19 @@ def _get_transformers(self, dtypes):
 
         Returns:
             dict:
-                mapping of field names and transformer instances.
+                A dict containing the ``sdtypes`` and ``transformers`` config for the
+                ``rdt.HyperTransformer``.
         """
         transformers = dict()
+        sdtypes = dict()
         for name, dtype in dtypes.items():
+            dtype = np.dtype(dtype).kind
             field_metadata = self._fields_metadata.get(name, {})
             transformer_template = field_metadata.get(
```

Comment:
does this take the metadata into account at all? I'm trying to figure out why it wasn't before and what has changed now. For example in the `student_placements` demo it wasn't making the `duration` column `categorical` even though the metadata said it was

Reply:
Old code was reading properly the `dtype` but was printing the wrong config which leaded to confusions as the `transformers` were updated after the `detect`. Now both `sdtypes` and `transformers` are being set corresponding to the `dtype` from the `metadata` with the `set_config` method.